core/Memory_Model/Storage/TaskRepositoryController.h

```diff
@@ -5,12 +5,13 @@
 #ifndef TASKMANAGER_CORE_MEMORY_MODEL_STORAGE_TASKREPOSITORYCONTOLLER_H_
 #define TASKMANAGER_CORE_MEMORY_MODEL_STORAGE_TASKREPOSITORYCONTOLLER_H_
 #include "TaskRepository.h"
+#include "Memory_Model/RepositoriesFactory/RepositoriesFactory.h"
 
 class TaskRepositoryController {
```

Comment:
You need virtual functions and destructor to provide any kind of testability and Dependency Injection.